src/6model/reprs/VMArray.c

```diff
@@ -227,36 +227,36 @@ void MVM_VMArray_at_pos(MVMThreadContext *tc, MVMSTable *st, MVMObject *root, vo
                 value->n64 = (MVMnum64)body->slots.n32[body->start + real_index];
             break;
         case MVM_ARRAY_U64:
-            if (kind != MVM_reg_int64)
-                MVM_exception_throw_adhoc(tc, "MVMArray: atpos expected int register");
+            if (kind != MVM_reg_uint64 && kind != MVM_reg_int64)
+                MVM_exception_throw_adhoc(tc, "MVMArray: atpos U64 expected int register, got %d instead", kind);
```

Comment:
How come this is the only one with the `kind` added to the message?

Reply:
It's the only one where I needed debugging help. Since my discussion with jnthn about the topic I've started committing changes like this.

Reply:
No objection to the addition here, it just seems out of place to only have it in the one case. But I think I still have a (yet another) WIP branch that adds context to error messages, the rest will probably get picked up whenever I get around to finished it up.